pkg/cmd/render/render.go

```diff
@@ -275,6 +296,55 @@ func mustReadTemplateFile(fname string) options.Template {
 	return options.Template{FileName: fname, Content: bs}
 }
 
+func discoverCIDRs(clusterConfigFileData []byte, renderConfig *TemplateData) error {
+	if err := discoverCIDRsFromNetwork(clusterConfigFileData, renderConfig); err != nil {
+		if err = discoverCIDRsFromClusterAPI(clusterConfigFileData, renderConfig); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func discoverCIDRsFromNetwork(clusterConfigFileData []byte, renderConfig *TemplateData) error {
```

Comment:
Why does this function take in a pointer to *TemplateData rather than returning []string, []string, error?

returning the addresses and assigning them separately will save us from creating a blank struct in testing [here](https://github.com/openshift/cluster-etcd-operator/pull/42/files#diff-ab2090bd61b2cf366b708a2ac6dd52adR109). IMO it is a better way of testing

Reply:
this is taken from upstream KAO